gap/generic/SnAnUnknownDegree.gi

```diff
@@ -700,49 +1009,52 @@ RECOG.ConstructSnAnIsomorphism := function(ri, n, stdGensAnWithMemory)
     if not RECOG.SatisfiesSnPresentation(ri, n, b, h) then
         return fail;
     fi;
-    xis := RECOG.ConstructXiSn(n, b, h);
+    if n >= 11 then
+        filter := RECOG.ConstructXiSn(n, b, h);
+    fi;
     for g in ri!.gensHmem do
-        gImage := RECOG.FindImageSn(ri, n, StripMemory(g),
-                                    b, h,
-                                    xis[1], xis[2]);
+        if n < 11 then
+            gImage := RECOG.FindImageSnAnSmallDegree(ri, n, StripMemory(g),
+                                                     filter[1][1], filter[1][2],
+                                                     filter[2]);
+        else
+            gImage := RECOG.FindImageSn(ri, n, StripMemory(g),
+                                        b, h,
+                                        filter[1], filter[2]);
+        fi;
         if gImage = fail then return fail; fi;
         slp := RECOG.SLPforSn(n, gImage);
         eval := ResultOfStraightLineProgram(slp, [h, b]);
         if not isequal(ri)(eval, StripMemory(g)) then return fail; fi;
     od;
     return rec(type := "Sn",
-               isoData := [[b, h], xis, n],
+               isoData := [[b, h], filter, n],
                slpToStdGens := SLPOfElms(stdGensSnWithMemory));
 end;
 
-# This method is an implementation of <Cite Key="JLNP13"/>. It is the main
-# function of SnAnUnknownDegree.
-# Note that it currently only works for 11 <= <A>n</A>. TODO: make it work with
-# smaller <A>n</A>, that is include fixes from Jonathan Conder's B.Sc.
-# Thesis "Algorithms for Permutation Groups", see PR #265.
+# ri : recognition node with group G,
+# T : integer, number of iterators
+# N : integer, upper bound for the degree of G
 #
-# From <Cite Key="JLNP13" Where="Theorem 1.1"/>:
-# RECOG.RecogniseSnAn is a one-sided Monte-Carlo algorithm with the following
-# properties. It takes as input a black-box group <A>G</A>, a natural number
-# <A>N</A> and a real number <A>eps</A> with 0 < <A>eps</A> < 1. If <A>G</A> is
-# isomorphic to An or Sn for some 9 <= <A>n</A> <= <A>N</A>, it returns with
-# probability at least 1 - <A>eps</A> the degree <A>n</A> and an
-# isomorphism from <A>G</A> to An or Sn.
-RECOG.RecogniseSnAn := function(ri, eps, N)
-    local T, foundPreImagesOfStdGens, constants, iterator, c, tmp, recogData, i;
-    T := Int(Ceil(Log2(1 / Float(eps))));
-    foundPreImagesOfStdGens := false;
-    constants := RECOG.ThreeCycleCandidatesConstants(1. / 4., N);
-    for i in [1 .. T] do
-        iterator := RECOG.ThreeCycleCandidatesIterator(ri, constants);
+# This is the main method used by RECOG.RecogniseSnAn and RECOG.RecogniseSnAnLazy.
```

Comment:
As far as I can tell, `RECOG.RecogniseSnAn` is not used anymore. Can we just remove it then, and perhaps rename `RECOG.RecogniseSnAnLazy` to `RECOG.RecogniseSnAn` ?